httomo/data/hdf/_utils/reslice.py

```diff
@@ -181,4 +192,7 @@ def single_sino_reslice(
     )
 
     if mpiutil.rank == 0:
+        assert recvbuf is not None
         return recvbuf.reshape((data_shape[0], data_shape[2]))
+    else:
+        return None
```

Comment:
Naively, the additions here don't look like they change the behaviour of the function:
- I think python functions implicitly `return None` if they reach the end of a function and haven't hit a `return` statement
- asserting that `recvbuf is not None` for rank 0 is sort of made redundant by the if block at line 174 which assigns `recvbuf` to something

If I get rid of those additions and run the `test_does_reslice_when_needed` test locally it passes for me, could you run locally on your end to confirm if those changes are needed or not?

If not, I suppose it's still fine to leave in, so feel free to take it or leave it.

```suggestion
    if mpiutil.rank == 0:
        return recvbuf.reshape((data_shape[0], data_shape[2]))
```

Reply:
Although mypy is now marks the whole function red as it explicitly missing the return statement...

Reply:
This is fair enough, I use `pyright` instead of `mypy` for type-checking (because I already use `pyright` as a language server anyway) and `pyright` doesn't complain to me about a missing return statement, they likely have different rules :shrug: